### PR TITLE
Add virtual destructors for CBLSWrapper and CBLSLazyWrapper

### PR DESCRIPTION
**Pull request by rkarthik2k21**
Add virtual destructors for CBLSWrapper and CBLSLazyWrapper. This prevents from potential memory leaks when derived classes are deleted (using the base pointer)

### CODE REVIEW

**Review by UdjinM6 — APPROVED** [review]
LGTM, utACK

File: src/bls/bls.h [hunk 1]
```diff
@@ -79,7 +79,9 @@ class CBLSWrapper
         std::swap(fLegacy, ref.fLegacy);
         return *this;
     }
-
+    
+    virtual ~CBLSWrapper() {}
+
```
Comment (UdjinM6) [inline src/bls/bls.h:84]:
```suggestion

    virtual ~CBLSWrapper() {}

```

File: src/bls/bls.h [hunk 2]
```diff
@@ -319,6 +321,8 @@ class CBLSLazyWrapper
     {
         *this = r;
     }
+
```
Comment (UdjinM6) [inline src/bls/bls.h:324]:
```suggestion

```

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

**Review by PastaPastaPasta — APPROVED** [review]
utACK for squash merge once CI is happy

**Review by PastaPastaPasta — APPROVED** [review]
re-utACK for squash merge

**Review by UdjinM6 — APPROVED** [review]
utACK

**Pull request merged**